### ISSUE TITLE
Dynamic Redirects with nested dynamic pages causing infinite loop.

### ISSUE THREAD

**Issue by eric-personal** [gladly-team/next-firebase-auth#440]
**Describe the bug**
The docs have a clear example of using Dynamic Redirects with regular routes but not one when using dynamic routes.
Example pages folder structure: 
- products
  - index.js
  - [productId].js

**Versions**

    "firebase": "^9.6.6",
    "firebase-admin": "^10.0.2",
    "next": "12.0.10",
    "next-firebase-auth": "^1.0.0-canary.6",

**To Reproduce**
Setup a dynamic page [productId].js as described in the folder structure above 
Set your redirect to happen in the dynamic page. 

 **[productId].js**
```
const ProductID = () => {
  return <div>Another sub product</div>;
};

export default withAuthUser({
  whenUnauthedAfterInit: AuthAction.REDIRECT_TO_LOGIN,
})(ProductID);
```
sign out of your app causing the redirect to login happen
This will cause an infinite loop in authPageURL: ({ ctx }) => {} causing your application to crash

The non dynamic page products index.js on the other hand works perfectly with dynamic redirects.
- products
  **- index.js**
```
const Products = () => {
  return <div>Products</div>;
};

export default withAuthUser({
  whenUnauthedAfterInit: AuthAction.REDIRECT_TO_LOGIN,
})(Products);
```

**Expected behavior**
Should allow dynamic pages to pass back their redirect path the same as regular pages do when using Dynamic Redirects.

**Comment by kmjennison** [comment]
Thanks for the very clear issue! This is an error in the example app's auth page setting—fixing now.

**Comment by kmjennison** [comment]
Closed in #442.
